### PR TITLE
Updates xesmf docs

### LINKED ISSUE #421: [Doc]: Update horizontal docs to outline how xCDAT extends xESMF
### Describe your documentation update

There might be a misconception that xCDAT only "wraps" xESMF and xgcm for horizontal and vertical regridding. 
Generally speaking, wrapping APIs involves encapsulating them with the goal of simplifying/streamlining usage rather than extending their functionality.

> "API wrappers can simplify the process of interacting with APIs. An API wrapper provides a way to access an API through a particular programming language or interface, which can help streamline the process of making API calls."
> -- https://guides.lib.utexas.edu/c.php?g=897091&p=6453521

### Solution
- [ ] Update `regridder.horizontal_xesmf()` docstring
  - [ ] Replace "Wraps" with "Extends"
  - [ ] List ways xCDAT extends xESMF (e.g., handles bounds)
- [ ] Update [horizontal regridding notebook](https://xcdat.readthedocs.io/en/stable/examples/horizontal-regridding.html) 
- [ ] Add information on when a user might want to `regrid2` or `xesmf` in docstrings (what are the differences?)
  - [ ] https://xcdat.readthedocs.io/en/stable/generated/xcdat.regridder.regrid2.Regrid2Regridder.html
  - [ ] https://xcdat.readthedocs.io/en/stable/generated/xcdat.regridder.xesmf.XESMFRegridder.html

### PR DESCRIPTION
**Pull request by jasonb5**
## Description

This PR updates the documentation for xESMF regridder.

- Closes #421

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] My changes generate no new warnings
- [ ] Any dependent changes have been merged and published in downstream modules

If applicable:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass with my changes (locally and CI/CD build)
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have noted that this is a breaking change for a major release (fix or feature that would cause existing functionality to not work as expected)

### CODE REVIEW

**Comment by jasonb5** [comment]
@pochedls @lee1043 @chengzhuzhang  @tomvothecoder 
Does anyone remember the difference between `regrid2` and `esmf`? I recall it had to do with `regrid2` treating the grid as flat but I can't remember any of the details.

**Comment by pochedls** [comment]
> @pochedls @lee1043 @chengzhuzhang @tomvothecoder Does anyone remember the difference between `regrid2` and `esmf`? I recall it had to do with `regrid2` treating the grid as flat but I can't remember any of the details.

@taylor13 - Is there a concise summary of regrid2 (for code docstrings)? It looks like CDAT had 

> To perform all the tasks required to regrid the input data into the ouput data in the latitude-level plane for all times

If I remember correctly, ESMF does not treat grid cell latitude bounds as constant in longitude and regrid2 does this? Is there a situation where a user would definitely want to use regrid2 over ESMF?

**Comment by jasonb5** [comment]
@taylor13 Thanks so much for all the great info.

File: xcdat/regridder/accessor.py [hunk 1]
```diff
@@ -108,10 +108,13 @@ def horizontal_xesmf(
...
-        Wraps the xESMF library providing access to regridding between
-        structured rectilinear and curvilinear grids.
+        Extends the xESMF library for horizontal regridding between structured
+        rectilinear and curvilinear grids.
 
-        Regrids ``data_var`` in dataset to ``output_grid``.
+        This method extends ``xESMF`` by automatically constructing the
+        ``xe.XESMFRegridder`` object, preserving source bounds, and generating
+        missing bounds. It regrids ``data_var`` in the dataset to
+        ``output_grid``.
```
Comment (tomvothecoder) [inline xcdat/regridder/accessor.py:117]: I updated the docstring for `horizontal_xesmf()`  using the new content from `XESMFRegridder`.

**Review by tomvothecoder — APPROVED** [review]
Hi @jasonb5, I made one minor edit which is noted in my PR review comment. Everything looks good. Thanks!

File: xcdat/regridder/accessor.py (hunk 1 above)
Comment (jasonb5) [reply xcdat/regridder/accessor.py:117]: Thanks, this was a good catch.

**Pull request merged**